enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,114 @@ any user-defined primary network.
-OVN-Kubernetes can validate that there are no pods still using this network. If there are pods still attached to this
-network, the network will not be removed.
+The main API extension here will be a namespace scoped network CRD as well a cluster scoped network CRD. 
+These CRDs will be registered by Cluster Network Operator (CNO) and managed by OVN-Kubernetes controller.
+
+#### CRDs
+
+Following the [CRDs for Managing Networks](#crds-for-managing-networks), two CRDs shall be introduced: 
+- Namespace scoped CRD - represent user request for creating namespace scoped OVN network.
+  - Shall be defined with `namespace` scope.
+  - Targeted for cluster admin and non-admin users, enable creating OVN network in a specific namespace.
+- Cluster scoped CRD - represent user request for creating cluster scoped OVN network, enables cross-namespace networking.
+  - Shall be defined with `cluster` scope
+  - Targeted for cluster admin users only, enable creating shared OVN network across multiple namespaces.
+
+Having a namespace-scope CRD targeted for admin and non-admin users, a cluster-scope CRD for admins only and utilizing RBAC
+mechanism, enable allowing non-admin users create OVN networks in namespaces they permitted to with no admin intervention, 
+without the risk of destabilizing the cluster nodes or break the cluster network.
+
+There should be a finalizer on the CRDs, so that upon deletion OVN-Kubernetes can validate that there are no pods still using this network.
+If there are pods still attached to this network, the network will not be removed.
+
+##### spec
+
+The CRDs spec defines as follows:
+
+| Field name     | Description                                                                                                                                                                                                                                                                                                                                                                                        | optional |
+|----------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|
+| Topology       | The topological configuration for the network. Must be one of `Layer2`, `layer3`.                                                                                                                                                                                                                                                                                                                  | No       |
```

Comment:
don't we need the persistent IPs knob here ? 

Only for L2 networks.

Reply:
The name of the feature gate in the openshift API is `PersistentIPsForVirtualization` .

I think maybe we could call this `VirtualizationAwareIPAM` or something like that.

Remember, to avoid the boolean, we can implement one, by adding an enum, whose values would be `Enabled` / `Disabled` . I don't think we'll have more than these 2 values... At least I don't more values for it in the future.

Reply:
@maiqueb please see my previous comment, Github acts weird and shows it been posted yesterday

Reply:
As we discussed offline, this feature makes sense on any network that spuns across different nodes. 

Currently only an admin can create a network, so I don't see that as a reason not to support this feature. You can say the same about subnets, or MTU.

> Is there a way to enable persistent IP internally?

No.

Reply:
I think we should include persistentIPs as a configuration. There's relatively low overhead. Just a constraint that topology type must be layer 2, which I think can be enforced by CEL. I'm not sure what the argument is against it.